### ISSUE TITLE
Query building

### ISSUE THREAD

**Issue by richlewis14** [brianc/node-postgres#1816]
Hi

I have a question around building a query in an efficient manner. At present to do a INSERT I have something like so

    async function saveFixture(f) {
      try {
        await pool.query('INSERT into fixtures (match_id, home_team, home_team_id, away_team, away_team_id, league_name) VALUES ($1, $2, $3, $4, $5, $6) RETURNING *', [f.match_id, f.home_team, f.home_team_id, f.away_team, f.away_team_id, f.league_name])
      } catch(e) {
        console.error('Error Occurred ', e);
        throw e;
      }
    }
    # f being an object like so { match_id: '1', home_team: 'Team 1', home_team_id: '11', away_team: 'Team 2', away_team_id: '12', league_name: 'League Name' }

So this is pretty inefficient right if wanting to insert multiple records into my database as each time is a separate call to the DB. Would i be better off having ALL the objects I want in an array first and then perform a multi row insert?

Something else that comes to mind is this statement `INSERT into fixtures (match_id, home_team, home_team_id, away_team, away_team_id, league_name)`. Each of my keys in my object correspond to the column name in the database, so is there a way of mapping these out rather than write each one (as I really have over 20, made this example shorter).

Could anyone assist with an example of how to make my code more efficient please using node-pg

Thank You

**Comment by sehrope** [comment]
Yes, one command that does 20 rows is going to be more efficient than 20 separate commands as the latter would have 19 more round trips to the DB.

There's a lot of approaches to make multiple inserts more efficient. If you already have all the data for all the rows then you can transform it into arrays and have the server handle all of them at once. Search the issues for this project for "array insert" and you'll find a couple examples. Here's one: https://github.com/brianc/node-postgres/issues/957#issuecomment-295583050

As for mapping column names to objects, you'll either have to do it yourself or have an ORM handle it. There's some options on the wiki: https://github.com/brianc/node-postgres/wiki/Extras.
